File: src/main/java/com/vaadin/demo/component/stepper/StepperBasic.java
package com.vaadin.demo.component.stepper;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.stepper.Step;
import com.vaadin.flow.component.stepper.Stepper;
import com.vaadin.flow.router.Route;

@Route("stepper-basic")
public class StepperBasic extends Div {

    public StepperBasic() {
        // tag::snippet[]
        Stepper stepper = new Stepper();
        
        Step step1 = new Step("Shipping address", "Enter your shipping details");
        Step step2 = new Step("Billing address", "Enter your billing details");
        Step step3 = new Step("Payment method", "Select payment option");
        Step step4 = new Step("Review order", "Review and confirm your order");
        
        stepper.add(step1, step2, step3, step4);
        // end::snippet[]

        add(stepper);
    }

    public static class Exporter extends DemoExporter<StepperBasic> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/stepper/stepper-basic.ts
import '@vaadin/stepper';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('stepper-basic')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-stepper>
        <vaadin-step
          label="Shipping address"
          description="Enter your shipping details"
        ></vaadin-step>
        <vaadin-step label="Billing address" description="Enter your billing details"></vaadin-step>
        <vaadin-step label="Payment method" description="Select payment option"></vaadin-step>
        <vaadin-step label="Review order" description="Review and confirm your order"></vaadin-step>
      </vaadin-stepper>
      <!-- end::snippet[] -->
    `;
  }
}
